optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -0,0 +1,385 @@
+        self.decoder = decoder
+        self.input_names = {input_key.name: idx for idx, input_key in enumerate(self.decoder.get_inputs())}
+        self.output_names = {output_key.name: idx for idx, output_key in enumerate(self.decoder.get_outputs())}
+
+    def forward(
+        self, input_ids, encoder_hidden_states, encoder_attention_mask, past_key_values=None
+    ) -> Seq2SeqLMOutput:
+
+        onnx_inputs = {
+            "input_ids": input_ids.cpu().detach().numpy(),
+            "encoder_attention_mask": encoder_attention_mask.cpu().detach().numpy(),
+        }
+
+        # Add the encoder_hidden_states inputs when needed
+        if "encoder_hidden_states" in self.input_names:
+            onnx_inputs["encoder_hidden_states"] = encoder_hidden_states.cpu().detach().numpy()
+
+        if past_key_values is not None:
+            # Flatten the past_key_values
+            past_key_values = [mha for past_key_value in past_key_values for mha in past_key_value]
+
+            # Add the past_key_values to the decoder inputs
+            for i, past_key_value in enumerate(past_key_values):
+                onnx_inputs[f"past_key_values_{i}.1"] = past_key_value.cpu().detach().numpy()
+
+        # Run inference
+        outputs = self.decoder.run(None, onnx_inputs)
+
+        # Tuple of length equal to : number of layer * number of multi-head attention
+        past_key_values = tuple(
+            torch.from_numpy(outputs[self.output_names[key]]) for key in self.output_names if "past_key_values" in key
+        )
+
+        # Tuple of tuple of length `n_layers`, with each tuple of length equal to the number on multi-head attention
+        # (number of self-attention and cross-attention per decoder layer)
+        num_mha = 4
+        past_key_values = tuple(past_key_values[i : i + num_mha] for i in range(0, len(past_key_values), num_mha))
+
+        return Seq2SeqLMOutput(
+            logits=torch.from_numpy(outputs[self.output_names["logits"]]), past_key_values=past_key_values
+        )
+
+
+class ORTModelForConditionalGeneration(torch.nn.Module, GenerationMixin):
+    def __init__(
+        self,
+        encoder_sess: onnxruntime.InferenceSession,
+        decoder_sess: onnxruntime.InferenceSession,
+        decoder_with_past_sess: onnxruntime.InferenceSession,
+        config: Dict,
+    ):
+        super().__init__()
+        self.encoder = ORTEncoder(encoder_sess)
+        self.decoder = ORTDecoder(decoder_sess)
+        self.decoder_with_past = ORTDecoder(decoder_with_past_sess)
+        self.config = PretrainedConfig.from_dict(config)
+        self.main_input_name = "input_ids"
+
+    def forward(
+        self,
+        input_ids: torch.LongTensor = None,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        decoder_input_ids: Optional[torch.LongTensor] = None,
+        encoder_outputs: Optional[Tuple[Tuple[torch.Tensor]]] = None,
+        past_key_values: Optional[Tuple[Tuple[torch.Tensor]]] = None,
```

Comment:
I would allow one to pass `np.ndarray` or pytorch Tensors, eventually converting PyTorch ones to numpy in this function.

Everything from there should be in "native" ORT framework. This is especially important if we are going to use `ORTValue`  which would be defined in this class.